osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -154,6 +154,9 @@ public abstract partial class DrawableHitObject : PoolableDrawableWithLifetime<H
         [Resolved(CanBeNull = true)]
         private IPooledHitObjectProvider pooledObjectProvider { get; set; }
 
+        [Resolved(CanBeNull = true)]
+        private IGameplayClock gameplayClock { get; set; }
```

Comment:
This doesn't need to be DI'd. It will be available to you if you soft-cast `Clock` to `IGameplayClock`.

Compare:

https://github.com/ppy/osu/blob/bb2f38d189b4a1a6ff888f5a74caf90ec60bf746/osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSliderBall.cs#L193